lib/YMLProcessing.groovy

```diff
@@ -0,0 +1,43 @@
+
+import nextflow.Nextflow
+import org.yaml.snakeyaml.Yaml
+
+class YMLProcessing {
+
+    /**
+     * Parses a YAML file to extract contrast data.
+     *
+     * @param ymlFile The YAML file to parse.
+     * @return A list of maps, each containing keys: id, variable, reference, and target.
+     */
+    public static List<Map<String, String>> parseContrastsFromYML(ymlFile) {
+        //if (!file(ymlFile).exists() || !file(ymlFile).canRead()) {
```

Comment:
Lets remove these commented lines, and instead add , "checkIfExists: true", to this other line here: https://github.com/nf-core/differentialabundance/blob/3893afe8521359802adb1efa1a4d7f2466d6778a/workflows/differentialabundance.nf#L151

Reply:
Sure! furthermore, there are some more controls after this block that should be taken care during the YML validation